src/record_syscall.cc

```diff
@@ -1997,6 +1997,7 @@ static void ptrace_verify_set_reg_set(RecordTask* t, size_t min_size,
   auto iov = t->read_mem(remote_ptr<typename Arch::iovec>(t->regs().arg4()));
   if (iov.iov_len < min_size) {
     syscall_state.emulate_result(-EIO);
+    return;
```

Comment:
Yikes.

Reply:
Yeah, we don't test the error paths at all in the ptrace emulation. We have many issues here beyond this one.